agent/app/agent.go

```diff
@@ -164,9 +167,15 @@ func (agent *ecsAgent) doStart(containerChangeEventStream *eventstream.EventStre
 		log.Criticalf("Error creating state manager: %v", err)
 		return exitcodes.ExitTerminal
 	}
-	// TODO:
-	// 1) Feature gate ECS cgroup root init
-	// 2) Update mocks and unit tests
+
+	// Conditionally create '/ecs' cgroup root
+	if agent.cfg.TaskCPUMemLimit {
+		err = agent.resource.Init()
+		if err != nil || resources.UnsupportedPlatform(err) {
```

Comment:
This check is redundant.

`|| resources.UnsupportedPlatform(err)` will never be true when the first term is false..